lib/auth/in_memory/types.ts

```diff
@@ -0,0 +1,51 @@
+import joi from 'joi';
+
+export type Callback<Data = any> = (err: Error | null | undefined, data?: Data) => void;
+
+export type Key = { access: string; secret: string };
```

Comment:
naming nitpick: as far as I've seen it the access key / secret key tuple is referred to as `credentials` usually